src/back/spv/writer.rs

```diff
@@ -1375,7 +1450,20 @@ impl Writer {
                 }
             }
             crate::Expression::GlobalVariable(handle) => self.global_variables[handle.index()].id,
-            crate::Expression::Constant(handle) => self.constant_ids[handle.index()],
+            crate::Expression::Constant(handle) => {
+                // If this is a const array, store it into the allocated variable
+                let composite_id = self.constant_ids[handle.index()];
+                match function.global_const_arrays.get(&handle) {
+                    Some(&var_id) => {
+                        block
+                            .body
+                            .push(Instruction::store(var_id, composite_id, None));
+                        var_id
```

Comment:
similarly to the other case, I don't think we can return `var_id` here, since it has a different type